File: buble_sort/bs.c

#include <stdio.h>
#include <stdlib.h>

void print_arr(int** arr, int arr_length) {
    printf("[");
    for (int i = 0; i < arr_length; i++) {
        if (i != 0)
            printf("\t");

        printf("%d", i[(*arr)]);
    }
    printf("]\n");

    return;
}

void bubble_sort (int **arr_p, int arr_length) {
    int sorted_index = arr_length;
    int *arr = *arr_p;

    while (sorted_index > 0) {
        int changed_index = 0;

        for (int i = 0; i < sorted_index - 1; i++) {
            if (arr[i] <= arr[i + 1]) {
                continue;
            }

            int temp = arr[i + 1];
            arr[i + 1] = arr[i];
            arr[i] = temp;

            changed_index = i + 1;
        }

        sorted_index = changed_index;
    }

    return;
}


int main (void) {
    int *num_arr;
    int num_input = 0;

    scanf("%d\n", &num_input);

    num_arr = malloc(sizeof(int) * num_input);
    for (int i = 0; i < num_input; i++) {
        int usr_input; 
        scanf("%d\n", &usr_input);

        i[num_arr] = usr_input;
    }

    print_arr(&num_arr, num_input);

    bubble_sort(&num_arr, num_input);

    print_arr(&num_arr, num_input);
    return 0;
}

File: buble_sort/bs.go
package main 

import (
    "fmt"
)

func BubbleSort (numArr *[]int) {
    var loopLength int = len(*numArr)

    for loopLength > 0 {
        var changedIndex int

        for i := 1; i < loopLength; i++ {
            if (*numArr)[i - 1] > (*numArr)[i] {
                temp := (*numArr)[i]
                (*numArr)[i] = (*numArr)[i - 1]
                (*numArr)[i - 1] = temp

                changedIndex = i
            }
        }

        loopLength = changedIndex
    }

    return
}

func main () {
    var lengthArr int
    fmt.Scanf("%d\n", &lengthArr)

    numArr := make([]int, 0, lengthArr)

    for i := 0; i < lengthArr; i++ {
        var input int
        fmt.Scanf("%d\n", &input)

        numArr = append(numArr, input)
    }

    fmt.Printf("%v\n", numArr)
    
    BubbleSort(&numArr)

    fmt.Printf("%v\n", numArr)

    return
}
